### PR TITLE
Fix few issues with random state of uninitialized variables.

### PR DESCRIPTION
**Pull request by Eyeke2**
While it's not fixing all cases of uninitialized variables, it fixes a few of them.

### CODE REVIEW

**Review by agalasso — CHANGES_REQUESTED** [review]
@Eyeke2 seeing a PR title declaring issues with uninitialized variables really grabbed my attention, but unless I am missing something, you appear to be are raising a false alarm with the PR title and description.

File: target.cpp [hunk 1]
```diff
@@ -234,6 +234,7 @@ TargetClient::TargetClient(wxWindow *parent) :
 {
     SetBackgroundStyle(wxBG_STYLE_PAINT);
 
+    memset(m_history, 0, sizeof(m_history));
```
[the remark this answers is not in the document]
Comment (Eyeke2) [reply target.cpp:237]: @agalasso I agree it's not an essential fix, just makes a cleaner code, IMHO. Initializing m_history to 0 doesn't do any harm, as it defines positions of points to be displayed in the Target window. Having ra=dec=0 will just draw a point at the center of the circle, which is better than drawing a random values of ra/dec, but I agree it shouldn't happen anyways. I personally don't like leaving uninitialized variables and prefer deterministic states.

File: myframe.cpp [hunk 2]
```diff
@@ -210,6 +210,7 @@ struct FileDropTarget : public wxFileDropTarget
 MyFrame::MyFrame()
     :
     wxFrame(nullptr, wxID_ANY, wxEmptyString),
+    pGuider(nullptr),
```
[the remark this answers is not in the document]
Comment (Eyeke2) [reply myframe.cpp:213]: @agalasso initializing pGuider to null should be weighted when considering there are places in code which test for pGuider to be non-null, such as inside functions like CalSanityDialog::OnRecal, CalstepDialog::CalstepDialog, send_catchup_events (just to name the few). I'm not so mush familiar with most of the PHD2 code, but if somehow, in any of these functions which test validity of pFrame->pGuider before it's being initialized on line 270, they will incorrectly try to access random memory.

[the remark this answers is not in the document]
Comment (d33psky) [reply myframe.cpp:213]: fwiw i agree with both sides here :) curious to see what will get chosen.

**Comment by agalasso** [comment]
Having un-referenced data uninitialized is a common idiom in this codebase and we're not going to start changing that now. Examples: any POD data type declared on the stack without an initializer, lots of stl and wxWidgets container types like std::vector, wx vector type instances, buffers allocated with malloc etc etc.
Seeing as we agree this pr is just implementing your personal preference in a couple places to have some unreferenced data initialized to zero, and not fixing any issue where uninitialized data is actually referenced, I'm going to close this.
If you have some cases where code references uninitialized data then by all means let's fix them.

**Comment by Eyeke2** [comment]
Ok, I don't mind. it just seems there are redundant places in code which test if pGuider is not a NULL pointer, which will remain questionable.

**Pull request closed without merging**